pkgs/os-specific/linux/zenergy/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchFromGitHub, kernel, kmod }:
+
+let
+  kernelDirectory = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
+  version = "a3e124477ee8197015481156b90100d49fa3cd84";
+in
+stdenv.mkDerivation {
+  inherit version;
+  pname = "zenergy";
+
+  src = fetchFromGitHub {
+    owner = "BoukeHaarsma23";
+    repo = "zenergy";
+    rev = version;
+    hash = "sha256-s1aoipSsLKO23kTd2uGxVUpqYSeitiz3UIoDIxg/Dj8=";
+  };
+
+  hardeningDisable = [ "format" "pic" ];
+
+  makeFlags = kernel.makeFlags ++ [
+    "KDIR=${kernelDirectory}"
+  ];
+
+  preBuild = ''
+    substituteInPlace Makefile --replace-fail "PWD modules_install" "PWD INSTALL_MOD_PATH=$out modules_install"
+  '';
+
+  nativeBuildInputs = [ kmod ] ++ kernel.moduleBuildDependencies;
+
+  installPhase = ''
+    make modules_install KDIR=${kernelDirectory}
+  '';
```

Comment:
It should be enough to set makeTarget and makeFlags like already done

Reply:
I searched in the repository for examples and tried to replicate, but in the `installPhase` the `KDIR` environment wasn't set on top of not running `modules_install` with `makeTargets = [ "modules_install" ]`.

Am I missing something?

Reply:
@SuperSandro2000 can you also give me an update on this?

Reply:
`  installTargets = [ "modules_install" ];` works

Reply:
Thank you for the heads-up! Can't believe I hadn't noticed that in the docs.